torch/_dynamo/variables/builtin.py

```diff
@@ -622,6 +622,17 @@ def call_isinstance(self, tx, arg, isinstance_type):
             unimplemented(
                 f"isinstance called on UserDefinedClass {arg} {isinstance_type}"
             )
+        # handle __instancecheck__ defined in user class
+        if (
+            isinstance(arg, variables.UserDefinedObjectVariable)
+            and "__instancecheck__" in isinstance_type.__class__.__dict__
+        ):
+            return variables.ConstantVariable(
+                isinstance_type.__class__.__instancecheck__(
+                    isinstance_type, arg.as_python_constant()
+                )
+            )
+
```

Comment:
Because dynamo uses ```issubclass```(L637) rather than ```isinstance``` directly, so ```__instancecheck__``` won't be called automatically.